apollo-router/src/apollo_studio_interop/mod.rs

```diff
@@ -189,11 +204,19 @@ impl UsageReportingGenerator<'_> {
         sorted_fragments.sort_by_key(|&(k, _)| k);
 
         sorted_fragments.into_iter().for_each(|(_, f)| {
-            result.push_str(&ApolloReportingSignatureFormatter::Fragment(f).to_string())
+            let formatter = SignatureFormatterWithAlgorithm {
+                formatter: &ApolloReportingSignatureFormatter::Fragment(f),
+                normalization_algorithm: self.normalization_algorithm,
+            };
+            result.push_str(&format!("{}", formatter))
```

Comment:
```suggestion
            write!(&mut result, "{formatter}").expect("infallible");
```
To reduce the intermediate string allocation